src/sidebar/util/build-thread.js

```diff
@@ -253,38 +227,63 @@ function hasVisibleChildren(thread) {
 }
 
 /**
- * @typedef Options
- * @prop {string[]} selected - List of currently-selected annotation ids, from
- *       the data store
+ * @typedef BuildThreadOptions
+ * @prop {Object.<string, boolean>} expanded - Map of thread id => expansion state
  * @prop {string[]} forcedVisible - List of $tags of annotations that have
  *       been explicitly expanded by the user, even if they don't
  *       match current filters
+ * @prop {string[]} selected - List of currently-selected annotation ids, from
+ *       the data store
  * @prop {(a: Annotation) => boolean} [filterFn] - Predicate function that
  *       returns `true` if annotation should be visible
  * @prop {(t: Thread) => boolean} [threadFilterFn] - Predicate function that
  *       returns `true` if the annotation should be included in the thread tree
- * @prop {Object.<string, boolean>} expanded - Map of thread id => expansion state
- * @prop {(a: Annotation, b: Annotation) => boolean} sortCompareFn - Less-than
- *       comparison function for sorting top-level annotations
- * @prop {(a: Annotation, b: Annotation) => boolean} replySortCompareFn - Less-than
- *       comparison function for sorting replies
+
+ * @prop {(a: Thread, b: Thread) => number} [sortCompareFn] - comparison
+ *       function for sorting top-level annotations
+ * @prop {(a: Thread, b: Thread) => number} [replySortCompareFn] - comparison
+ *       function for sorting replies
+ */
+
+/**
+ * Default top-level thread sorting function if none provided via options.
+ *
+ * Sort by local application-applied `$tag`
+ *
+ * @param {Thread} a
+ * @param {Thread} b
+ * @return {number}
  */
+const defaultSortCompareFn = (a, b) => {
+  if (a.annotation && b.annotation) {
+    if (a.annotation.$tag < b.annotation.$tag) {
+      return -1;
+    } else if (a.annotation.$tag > b.annotation.$tag) {
+      return 1;
+    }
+  }
+  return 0;
+};
```

Comment:
It doesn't look like `defaultSortCompareFn` is used in the actual application. Perhaps we could simply re-use the same sorting function as we use for replies as a default, or simply not have a default at all and always require it to be set.